### PR TITLE
TEST: adding stream-table join

### PR DESCRIPTION
**Pull request by bluemonk3y**
#375 - adding missing integration tests

### CODE REVIEW

File: ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java [hunk 1]
```diff
@@ -17,8 +17,8 @@
 package io.confluent.ksql;
 
 import io.confluent.ksql.metastore.MetaStore;
-import io.confluent.ksql.util.KafkaTopicClientImpl;
-import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.*;
```
Comment (dguy) [inline ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java]: you probably need to change your import settings?

File: ksql-engine/src/test/java/io/confluent/ksql/integration/IntegrationTestHarness.java [hunk 2]
```diff
@@ -77,16 +78,24 @@ public void createTopic(String topicName, int numPartitions, short replicatonFac
...
       result.put(key, recordMetadataFuture.get(TEST_RECORD_FUTURE_TIMEOUT_MS, TimeUnit.MILLISECONDS));
     }
     producer.close();
 
     return result;
   }
 
+  private ProducerRecord<String, GenericRow> buildRecord(String topicName, Long timestamp, Map.Entry<String, GenericRow> recordEntry, String key) {
+    ProducerRecord<String, GenericRow> producerRecord;
+    if (timestamp == null) {
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/integration/IntegrationTestHarness.java]: You don't need this. Can just pass in null for the timestamp

File: ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ public void shouldAggregateTumblingWindow() throws Exception {
 
     // not really required - but lets mess with  ms
     Thread.sleep(100);
-    testHarness.publishTestData(topicName, dataProvider);
+    testHarness.publishTestData(topicName, dataProvider, null);
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java]: Can you get rid of the sleeps in this class now?

File: ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java (hunk 1 above)
Comment (bluemonk3y) [reply ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java]: err - yep - will do

File: ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java (hunk 3 above)
Comment (bluemonk3y) [reply ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java]: good spot

**Comment by bluemonk3y** [comment]
@dguy  - how does the look - we now control time and the future.

File: ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java [hunk 4]
```diff
@@ -64,8 +67,7 @@ public void after() throws Exception {
   public void shouldAggregateTumblingWindow() throws Exception {
 
     // not really required - but lets mess with  ms
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java]: Are these comments still relevant?

Comment (bluemonk3y) [reply ksql-engine/src/test/java/io/confluent/ksql/integration/WindowingIntTest.java]: good point!

**Comment by bluemonk3y** [comment]
retest this please

**Pull request merged**